lib/csv_row_model/export.rb

```diff
@@ -25,6 +30,34 @@ def to_row
       attributes.values
     end
 
+
+    # @return [Array] an array of rows, where if cell is row_name, it's parsed into the header_match
+    #                 and everything else is return as is.
+    def to_template
+      rows_temaplate.map do |row|
+        result = []
+        row.each do |cell|
+          if is_row_name? cell
+            header_matchs = self.class.options(cell)[:header_matchs]
+            result << "#{header_matchs ? header_matchs.first : self.class.format_header(cell)}"
+            result << "#{attributes[cell]}"
+          else
+            result << cell.to_s
+          end
+        end
+        result
+      end
+    end
+
+    # Safe to override
+    #
+    # @return [Array<Array>] an array of arrays, where every represents a row and every row
+    #                        can have strings and row_name (column_name). By default,
+    #                        returns a row_name for every row
+    def rows_temaplate
+      @rows_temaplate ||= self.class.row_names.map{ |row_name| [row_name]}
+    end
+
```

Comment:
let's move the changes in this file to a `SingleModel` module to separate this logic from the rest.